lifecycle-operator/controllers/common/helperfunctions.go

```diff
@@ -40,10 +41,13 @@ func GetOldStatus(name string, statuses []klcv1beta1.ItemStatus) apicommon.Keptn
 	return oldstatus
 }
 
-func CopyMap[M1 ~map[K]V, M2 ~map[K]V, K comparable, V any](dst M1, src M2) {
-	for k, v := range src {
-		dst[k] = v
-	}
+func MergeMaps[M1 ~map[K]V, K comparable, V any](map1 M1, map2 M1) M1 {
+	merged := make(M1, len(map1)+len(map2))
+	// we copy the map2 first, so the values in the overlapping
+	// properties are set from map1 in the resulting map
```

Comment:
nit: can we flip this so we have a similar behaviour to `maps.Copy()`? there the second parameter seems to take precedence

Reply:
sure, I can adapt it

Reply:
adapted